src/ode_solver/ode_solver_base.cpp

```diff
@@ -13,10 +13,7 @@ ODESolverBase<dim,real,MeshType>::ODESolverBase(std::shared_ptr< DGBase<dim, rea
         , pcout(std::cout, dealii::Utilities::MPI::this_mpi_process(mpi_communicator)==0)
         , refine_mesh_in_ode_solver(true)
         {
-            meshadaptation = std::make_unique<MeshAdaptation<dim,real,MeshType>>(all_parameters->mesh_adaptation_param.critical_residual_val,
-                                                                                 all_parameters->mesh_adaptation_param.total_refinement_steps,
-                                                                                 all_parameters->mesh_adaptation_param.refinement_fraction,
-                                                                                 all_parameters->mesh_adaptation_param.coarsening_fraction);
+            meshadaptation = std::make_unique<MeshAdaptation<dim,real,MeshType>>(dg);
```

Comment:
Since dg is being passed here, is it necessary to pass dg every time adapt_mesh is called on line 156? Maybe MeshAdaptation could contain a pointer to dg.

Reply:
Having dg as a class member gave an error of pointer corruption earlier. However, it's working well now, so I guess using virtual destructor fixed it. The new commit has dg as a member in both MeshAdaptation  and MeshErrorEstimate.